tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/util/PingThread.java

```diff
@@ -84,15 +84,17 @@ protected void ping() {
 
 				URL pingUrl = new URL(url);
 				URLConnection conn = pingUrl.openConnection();
+
 				((HttpURLConnection)conn).setInstanceFollowRedirects(false);
-				((HttpURLConnection)conn).getResponseCode();
+				int code = ((HttpURLConnection)conn).getResponseCode();
 
 				// ping worked - server is up
-				if (!stop) {
+				if (!stop && code == 200) {
```

Comment:
Checking the code makes sense but are there other valid response codes that would mean that Liferay is up?